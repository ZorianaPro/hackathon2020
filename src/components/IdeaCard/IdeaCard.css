.IdeaCard {
    background: rgba(255,255,255,0.7);
    padding: 0.5em;
    display: inline-block;
    margin: 1.5em;
    border-radius: 9px;
    box-shadow: 0 0 30px 0 rgba(0, 0, 0, 0.1);
    height: 22em;
    width: 22em;
    transition: box-shadow 200ms ease-in;
}

.IdeaCard:hover {
    box-shadow: 0 15px 30px 0 rgba(0, 0, 0, 0.3);
}

.IdeaCard .IdeaCard-Container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;
    height: 100%;
    position: relative;
}

.IdeaCard .IdeaCard-Info {
    display: flex;
    flex-direction: column;
    padding: var(--fontSize36) 1em 1em 1em;
}

.IdeaCard .IdeaCard-Info .IdeaCard-Info-Name {
    position: relative;
    font-size: var(--fontSize24);
    font-weight: var(--fontWeightPoppinsBold);
    font-stretch: normal;
    font-style: italic;
    line-height: var(--lineHeight);
    letter-spacing: normal;
    color: var(--colorSpaceCadet);
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
}

.IdeaCard .IdeaCard-Info .IdeaCard-Info-Name:after {
    content: " ";
    position: absolute;
    bottom: 0;
    left: 0;
    width: 80px;
    height: 4px;
    background-color: var(--colorRadicalRed);
}

.IdeaCard .IdeaCard-Info .IdeaCard-Info-Description {
    font-size: var(--fontSize16);
    font-stretch: normal;
    padding-top: 1em;
    line-height: var(--lineHeight);
    letter-spacing: normal;
    color: var(--colorSpaceCadet);
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 4;
}

.IdeaCard hr {
    height: 1px;
    width: 100%;
    background-color: #ecf0f3;
}

.IdeaCard .IdeaCard-Actions {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-top: 2em;
    align-items: center;
    border-top: 1px solid #ecf0f3;
    padding-top: 0.5em;
}

.IdeaCard .IdeaCard-Actions .IdeaCard-More {
    padding-top: 1em;
    cursor: pointer;
    text-transform: uppercase;
    text-decoration: none;
    transition: all ease-in 100ms;
}

.IdeaCard .IdeaCard-Actions .IdeaCard-More:hover {
   text-decoration: underline;
}

.IdeaCard .IdeaCard-Actions .Idea-Actions-Join {
    padding-top: 1em;
    cursor: pointer;
    text-decoration: none;
    transition: all ease-in 100ms;
}

.IdeaCard-Actions .Button--isSecondaryTransparent:nth-child(1) {
    width: 60%;
    border-bottom-left-radius: 9px;
    margin-right: 0.5em;
}

.IdeaCard-Actions .Button--isSecondaryTransparent:nth-child(2) {
    border-bottom-right-radius: 9px;
    width: 37%;
}

.IdeaCard-Actions svg {
    width: 24px;
    height: 24px;
    margin-right: 1em;
}
