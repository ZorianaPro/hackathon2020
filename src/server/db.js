const mongoose = require('mongoose');

const connect = () => {
  return mongoose.connect(
    'mongodb+srv://zoriana:123123321@hackathon2020-673vc.mongodb.net/hack?retryWrites=true&w=majority',
    { useNewUrlParser: true },
    function(err) {
      if (err) {
        throw err;
      }
    }
  );
};

module.exports = {
  connect
};