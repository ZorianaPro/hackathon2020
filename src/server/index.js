const Koa = require("koa");
const BodyParser = require("koa-bodyparser");
const Logger = require("koa-logger");
const cors = require("koa-cors");
const Router = require("koa-router");
const HttpStatus = require("http-status");

const app = new Koa();

const PORT = process.env.PORT || 8082;

app.use(BodyParser());
app.use(Logger());

app.use(cors());

const router = new Router();

router.get("/book", async (ctx, next) => {
  const books = [
    "Learn somehting man",
    "Fluent Python",
    "Pro Python",
    "The Go programming language",
  ];

  ctx.status = HttpStatus.OK;
  ctx.body = books;
  await next();
});

app.use(router.routes()).use(router.allowedMethods());

app.listen(PORT, function () {
  console.log(
    "==> ðŸŒŽ  Listening on port %s. Visit http://localhost:%s/",
    PORT,
    PORT
  );
});
