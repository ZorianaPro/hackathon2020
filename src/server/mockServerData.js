const teams = [
	{
		id: 1,
		members: [
			{
				id: 1,
				firstName: 'Drfg',
				lastName: 'Drfg',
				position: 'Software Engineer',
				email: 'dfgdg@dff.fff'
			}, {
				id: 2,
				firstName: 'Drfg',
				lastName: 'Drfg',
				position: 'Software Engineer',
				email: 'dfgdg@dff.fff'
			}, {
				id: 3,
				firstName: 'Drfg',
				lastName: 'Drfg',
				position: 'Software Engineer',
				email: 'dfgdg@dff.fff'
			}
		]
	}, {
		id: 2,
		members: [
			{
				id: 1,
				firstName: 'Drfg',
				lastName: 'Drfg',
				position: 'Software Engineer',
				email: 'dfgdg@dff.fff'
			}, {
				id: 2,
				firstName: 'Drfg',
				lastName: 'Drfg',
				position: 'Software Engineer',
				email: 'dfgdg@dff.fff'
			}, {
				id: 3,
				firstName: 'Drfg',
				lastName: 'Drfg',
				position: 'Software Engineer',
				email: 'dfgdg@dff.fff'
			}
		]
	}
];

const ideas = [
	{
		id: 1,
		name: 'Idea 1 name',
		description: 'Any long description Any long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long descriptionAny long description',
		teamId: 1
	},
	{
		id: 2,
		name: 'Idea 2 name',
		description: 'Any long description',
		teamId: 2
	}
];

export {
	ideas,
	teams
};