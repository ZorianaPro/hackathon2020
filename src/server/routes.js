const Router = require("koa-router");
const HttpStatus = require("http-status");
const Idea = require('./controllers/idea');

const router = new Router();

router.get("/ideas", async (ctx, next) => {
	const ideas = await Idea.showAll();
	ctx.status = HttpStatus.OK;
	ctx.body = ideas;
	await next();
});

router.post("/idea", async (ctx, next) => {
	const content = ctx.request.body;
	const idea = await Idea.create(content);

	ctx.status = HttpStatus.OK;
	ctx.body = idea.toJSON();
	await next();
});

router.post('/join', async (ctx, next) => {
	const content = ctx.request.body;
	const idea = await Idea.addTeamMember(content.id, content.member);
	ctx.status = HttpStatus.OK;
	ctx.body = idea.toJSON();
	await next();
});

router.get('/ideas/delete', async (ctx, next) => {
	const idea = await Idea.deleteAll();
	ctx.status = HttpStatus.OK;
	await next();
});

module.exports = router;