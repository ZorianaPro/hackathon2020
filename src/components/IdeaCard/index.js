import React, { useState, useEffect, useCallback } from 'react';
import { teams } from '../../server/mockServerData';
import Team from '../Team';
import * as ideaOverlay from "../IdeaInfoOverlay/actions";
import * as joinOverlay from "../JoinIdeaOverlay/actions";
import  './IdeaCard.css';
import { useDispatch } from "react-redux";

const IdeaCard = ({
  name,
	description,
	team,
	id,
	dispatch
}) => {


	dispatch = useDispatch();

	const joinIdeaOverlay = useCallback(() => {
			dispatch(joinOverlay.open({
				id: id
			}))
		}, [id, dispatch]);

	const openIdeaOverlay = useCallback(() => {
		dispatch(ideaOverlay.open({
			id: id,
			name: name,
			description: description,
			team: team
		}))
	}, [id, name, description, team, dispatch]);

	return (
		<div className="IdeaCard" key={id}>
			<div className="IdeaCard-Container">
				<div className="IdeaCard-Info">
					<div className="IdeaCard-Info-Name">
						{name}
					</div>
					<div className="IdeaCard-Info-Description">
						{description}
					</div>
				</div>
				<div className="IdeaCard-Actions">
					{
						team
						&& <Team full={false}
						         team={team}/>
					}
					<div className="Idea-Actions-Join"
					     onClick={() => joinIdeaOverlay()}>
						+ join
					</div>
					<div className="IdeaCard-More"
					     onClick={() => openIdeaOverlay()}>
						MORE
					</div>
				</div>
			</div>
		</div>
	)
};

export default IdeaCard;