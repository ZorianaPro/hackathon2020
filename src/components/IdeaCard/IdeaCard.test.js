import React from 'react';
import { shallow, mount } from 'enzyme';
import IdeaCard from './';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import { teams } from '../../server/mockServerData'

describe('IdeaCard', () => {
	const initialState = {
		open: false,
		ideaContent: {
			id: '',
			name: '',
			description: '',
			team: {}
		}
	};
	const mockStore = configureStore();
	let store = mockStore(initialState);

	it('renders without crashing', () => {
		expect(mount.bind(shallow, <Provider store={store}><IdeaCard /></Provider>)).not.toThrow();
	});

	it('renders as expected', () => {
		expect(mount(<Provider store={store}><IdeaCard /></Provider>)).toMatchSnapshot();
	});

	describe('given props', () => {
		let card;
		const props = {
			id: 12,
			name: 'Name',
			description: 'fddf gfdg',
			teamId: 1
		};

		beforeEach(() => {
			store = mockStore(initialState);
			card = mount(<Provider store={store}><IdeaCard { ...props }/></Provider>);
		});
		it('renders content', () => {
			expect(card.find('.IdeaCard-Info-Name').text()).toBe(props.name);
			expect(card.find('.IdeaCard-Info-Description').text()).toBe(props.description);
		});
	});
});