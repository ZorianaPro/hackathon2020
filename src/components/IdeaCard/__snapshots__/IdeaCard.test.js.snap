// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard renders as expected 1`] = `
<div
  className="IdeaCard"
>
  <div
    className="IdeaCard-Container"
  >
    <div
      className="IdeaCard-Info"
    >
      <div
        className="IdeaCard-Info-Name"
      />
      <div
        className="IdeaCard-Info-Description"
      />
    </div>
    <div
      className="IdeaCard-Actions"
    >
      <div
        className="Team"
      >
        <div
          className="Team-Members"
        />
        <div
          className="Team-Join"
          onClick={[Function]}
        >
          + join
        </div>
      </div>
      <div
        className="IdeaCard-More"
        onClick={[Function]}
      >
        MORE
      </div>
    </div>
  </div>
</div>
`;
